### PR TITLE
fix: UnhandledPromiseRejection errors

### PR DESCRIPTION
**Pull request by fboucquez**
Exception handling simplification in http calls
Defer when loading node info metadata
--unhandled-rejections=strict in tests

Fixes https://github.com/nemtech/symbol-sdk-typescript-javascript/issues/781

### CODE REVIEW

File: test/infrastructure/RepositoryFactory.spec.ts [hunk 1]
```diff
@@ -70,6 +72,17 @@ describe('RepositoryFactory', () => {
         expect(repositoryFactory.createFinalizationRepository()).to.be.not.null;
     });
 
+    it('Raise error without unhandled-rejections', async () => {
+        const url = 'https://www.google.com';
+        const repositoryFactory = new RepositoryFactoryHttp(url);
+        try {
+            await repositoryFactory.createNodeRepository().getNodeHealth().toPromise();
+            expect(true).to.be.false;
```
Comment (yilmazbahadir) [inline test/infrastructure/RepositoryFactory.spec.ts:80]: Should we use [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/) or a similar promise assertion library as a better practice in general for promise assertions?

File: test/infrastructure/RepositoryFactory.spec.ts [hunk 2]
```diff
@@ -70,6 +72,17 @@ describe('RepositoryFactory', () => {
         expect(repositoryFactory.createFinalizationRepository()).to.be.not.null;
     });
 
+    it('Raise error without unhandled-rejections', async () => {
+        const url = 'https://www.google.com';
```
Comment (yilmazbahadir) [inline test/infrastructure/RepositoryFactory.spec.ts:76]: Any other way to achieve this without relying on a remote call? I think we can use `sinon` to stub the `getNodeHealth`method to throw an error.

File: test/infrastructure/RepositoryFactory.spec.ts (hunk 1 above)
Comment (fboucquez) [reply test/infrastructure/RepositoryFactory.spec.ts:90]: Thanks! chai as promises looks nice. Although the try-catch could be more end-user-style assertion? And I wouldn't add another lib just yet. Let's consider it for the future.

File: test/infrastructure/RepositoryFactory.spec.ts (hunk 2 above)
Comment (fboucquez) [reply test/infrastructure/RepositoryFactory.spec.ts]: TS SDK uses ts-mockito. I've mocked the low-level generated NodeRoutesApi reflecting how it rejects responses with the current mock lib. The (nodeRepository as any).nodeRoutesApi doesn't look nice, but well.

**Review by rg911 — APPROVED** [review]

**Review by yilmazbahadir — APPROVED** [review]

**Pull request merged**